packages/core/src/utils/buildMetadata.ts

```diff
@@ -0,0 +1,26 @@
+import type { Options, SdkInfo } from '@sentry/types';
+
+import { SDK_VERSION } from '../version';
+
+const PACKAGE_NAME_PREFIX = 'npm:@sentry/';
+
+/**
+ * A builder for the SDK metadata in the options for the SDK initialization.
+ *
+ * @param options sdk options object that gets mutated
+ * @param sdkName name of the SDK (e.g. 'nextjs')
+ * @param packageNames list of package names (e.g. ['nextjs', 'react'])
+ */
+export function buildMetadata(options: Options, sdkName: string, packageNames: string[]): void {
```

Comment:
l: I would suggest either making this function pure (ie. making it return a new obj) or call it something like `applyMetadata()` because it mutates the passed options object.

Reply:
If we move forward with this (see other comments), I think I'm gonna go with `applyMetadata` for bundle size reasons